src/modalities/dataloader/large_file_lines_reader.py

```diff
@@ -56,21 +55,6 @@ def __getitem__(self, key: int | slice) -> str | List[str]:
         return self.__read_from_raw_file(offset, sample_length_in_bytes)
 
     def __read_from_raw_file(self, offset: int, sample_length_in_bytes: int) -> str:
-        def safe_decoder(byte_char):
-            try:
-                # TODO: verify why iso-8859-1 was necessary here in the path.
-                #   Maybe there was an issue with the actual loading of the jsonl-files
-                c = byte_char.decode("utf8")
-            except Exception as exception:
-                c = ""
-                warnings.warn(f'Encountered invalid char: "{byte_char}".')
-                warnings.warn(f"Encountered problem: {exception}")
-            return c
-
-        string = (
-            np.memmap(self.raw_data_path, mode="r", offset=offset, shape=(sample_length_in_bytes,)).view("S1").tolist()
-        )
-        decoded_string = []
-        for c in string:
-            decoded_string.append(safe_decoder(c))
-        return "".join(decoded_string)
+        f = self.raw_data_path.open(encoding=DEFAULT_ENCODING)
```

Comment:
Since the access here is controlled by the external component calling `__getitem__`, this pool needs a setup there. However only the LFLR knows where the files point to. Therefore one would sacrifice readability (by passing around FileStreams pointing to paths, which are private knowledge to another component) for performance, while the performance boosts for keeping a file descriptor active is actually not that high.